sdk/src/asset_handlers/riff_io.rs

```diff
@@ -541,90 +541,85 @@ impl RemoteRefEmbed for RiffIO {
     ) -> Result<()> {
         match embed_ref {
             crate::asset_io::RemoteRefEmbedType::Xmp(manifest_uri) => {
-                match self.riff_format.as_ref() {
-                    "avi" | "wav" | "webp" => {
-                        if let Some(curr_xmp) = self.read_xmp(input_stream) {
-                            let mut new_xmp = add_provenance(&curr_xmp, &manifest_uri)?;
-                            if new_xmp.len() % 2 == 1 {
-                                // pad if needed to even length
-                                new_xmp.push(' ');
-                            }
-
-                            let top_level_chunks = {
-                                let mut reader = CAIReadWrapper {
-                                    reader: input_stream,
-                                };
-                                Chunk::read(&mut reader, 0)?
-                            };
-
-                            if top_level_chunks.id() != RIFF_ID {
-                                return Err(Error::InvalidAsset("Invalid RIFF format".to_string()));
-                            }
+                if let Some(curr_xmp) = self.read_xmp(input_stream) {
```

Comment:
The reason for the old code is to make sure we confirmed that those formats did indeed support XMP (not all formats do).  I think this change is OK since those are currently the only RIFF based asset types we support.